docs/plugins/stable-plugins.asciidoc

```diff
@@ -0,0 +1,328 @@
+takes care of it. However you build the jar, once you have it, you need to
+write a plugin descriptor file. For a stable plugin, it should be called
+`stable-plugin.descriptor`.
+
+["source","properties"]
+------------------------
+description=a hello world stable analysis plugin
+version=1.0-SNAPSHOT
+name=hello-world-plugin
+java.version=17
+elasticsearch.version=8.7.0
+------------------------
+
+This and the jar should be compressed into a single ".zip" file.
+
+This file may then be installed on a node using the elasticsearch-plugin
+tool.
+
+If you are using the gradle plugin, you can compile the code, build the jar,
+generate the properties file, and put it all in a zip file just by running
+the task `bundlePlugin`. The plugin artifact will be written to
+`build/distributions`.
+
+A sample Gradle build file for using the Gradle plugin:
+
+["source","gradle"]
+-------------------------
+buildscript {
+    ext.elasticsearchVersion = '8.7.0-SNAPSHOT'
+    ext.luceneVersion = '9.4.1'
+    repositories {
+        // TODO: can uncredentialed users pull from the snapshot repository?
+        mavenLocal()
+        mavenCentral()
+    }
+    dependencies {
+        classpath "org.elasticsearch.gradle:build-tools:${elasticsearchVersion}"
+    }
+}
+
+apply plugin: 'java'
+apply plugin: 'elasticsearch.stable-esplugin'
+apply plugin: 'elasticsearch.yaml-rest-test'
+
+group 'org.example'
+version '1.0-SNAPSHOT'
+
+repositories {
+    ext.log4jVersion = '2.19.0'
+    mavenCentral()
+    mavenLocal()
+}
+
+dependencies {
+    compileOnly "org.elasticsearch:elasticsearch-plugin-api:${elasticsearchVersion}"
+    compileOnly "org.elasticsearch:elasticsearch-plugin-analysis-api:${elasticsearchVersion}"
+    compileOnly "org.apache.lucene:lucene-analysis-common:${luceneVersion}"
+
+    testImplementation "org.elasticsearch:elasticsearch-plugin-api:${elasticsearchVersion}"
+    testImplementation "org.elasticsearch:elasticsearch-plugin-analysis-api:${elasticsearchVersion}"
+    testImplementation "org.apache.lucene:lucene-analysis-common:${luceneVersion}"
+
+    testImplementation ('junit:junit:4.13.2'){
+        exclude group: 'org.hamcrest'
+    }
+    testImplementation 'org.hamcrest:hamcrest:2.2'
+}
+
+esplugin {
+    name 'hello-world-plugin'
+    description 'a hello world stable analysis plugin'
+}
+-------------------------
+
+[discrete]
+=== Unit testing
+
+Unit tests may go under the `src/test` directory. You will have to add
+dependencies for your preferred testing framework.
+
+[discrete]
+=== YAML REST tests
+
+If you are using the Elasticsearch stable api plugin for Gradle, you can
+use Elasticsearch's YAML Rest Test framework. This framework allows you to
+load your plugin in a running test cluster and issue real REST API queries
+against it. The full syntax for this framework is beyond the scope of this
+tutorial, but there are many examples in the Elasticsearch repository.
```

Comment:
we can also refer users to plugins/examples/stable-plugin where there is a sample yaml test